src/main/kotlin/org/rust/ide/annotator/RsFileErrorAnnotator.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator
+
+import com.intellij.codeInsight.daemon.impl.HighlightRangeExtension
+import com.intellij.ide.annotator.AnnotatorBase
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import org.rust.cargo.project.model.cargoProjects
+import org.rust.cargo.project.model.impl.allTargets
+import org.rust.lang.core.psi.RsFile
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.RsInnerAttr
+import org.rust.lang.core.psi.RsVisitor
+import org.rust.lang.core.psi.ext.descendantsOfType
+import org.rust.lang.utils.RsDiagnostic
+import org.rust.lang.utils.addToHolder
+
+class RsFileErrorAnnotator : AnnotatorBase(), HighlightRangeExtension {
```

Comment:
Let's better convert it into inspection (see `RsLocalInspectionTool` as base class).
It preferable for users because they can disable any inspection via settings. It allows to provide easy workaround in case of annoying false positive annotatons